Add location service of postnl

Jomaanro opened the pull request:
I've added the basic functionality for the location service provided by PostNL. It will locate the nearest dropoff point from a zipcode. In the future I will add the remaining two functions (locations in area and location info)

Jomaanro committed:
Add location

Jomaanro committed:
Location service updates

Jomaanro committed:
Update composer for local usage

Jomaanro committed:
Undo last commit and fix description

Jomaanro committed:
Remove function for now

Jomaanro committed:
Remove unused new file and remove some comment

slokhorst commented on src/Postnl.php:
```diff
@@ -315,6 +315,20 @@ public function currentStatus($barcode)
+     */
+    public function GetNearestLocation($PostcalCode, $AllowSundaySorting = "false", $DeliveryDate = null) {
+        $message = new ComplexTypes\Message;
+        $location = new ComplexTypes\Location($PostcalCode, $AllowSundaySorting, $DeliveryDate);
```
Typo: `PostcalCode` (also 2 lines below)

slokhorst commented:
Looks really useful!

A small style note: the project uses lower case (first letter) variables everywhere, but this pull uses upper case. Would be nice to be consistent.

Jomaanro committed:
Correctly camelCase functions, update comment blocks

Jomaanro commented:
Thanks for reviewing. Updated the functions to use correct camelCasing. I see that variables do have uppercase in all the other files so I left it like that. Also updated some more comment sections.

slokhorst commented:
Ah sorry you're right, I only looked at `Postnl.php`. Looks good!

ameenross commented:
Yeah, I kinda hated the camelcased variables. I think I had to do it because of how SoapClient serializes an object to XML.

Jomaanro committed:
Add Timeframe service

Jomaanro committed:
Add house number variable

Jomaanro committed:
Only daytime option

reijervano commented:
Nice work! Will you accept the request soon?

ameenross commented on src/ComplexTypes/GetNearestLocationsRequest.php:
```diff
@@ -0,0 +1,81 @@
+     * @param Message $Message
+     * @param Location $Location
+     */
+    public function __construct(
```
The parameters should be a single line

ameenross commented on src/ComplexTypes/GetNearestLocationsRequest.php:
```diff
@@ -0,0 +1,81 @@
+     */
+    public function setMessage($Message)
+    {
+        $this->Message = $Message;
```
``` php
return $this;
```

For chaining methods

ameenross commented on src/ComplexTypes/Location.php:
```diff
@@ -0,0 +1,149 @@
+    /**
+     * @var array $DeliveryOptions
+     */
+    protected $DeliveryOptions = array();
```
Use short array syntax. I require 5.4 anyway

ameenross commented on src/ComplexTypes/Location.php:
```diff
@@ -0,0 +1,149 @@
+    public function __construct($Postalcode, $AllowSundaySorting = "false", $DeliveryDate = null)
+    {
+        $this->setAllowSundaySorting($AllowSundaySorting);
+        $this->setDeliveryDate($DeliveryDate ?: date('d-m-Y',strtotime('next monday')));
```
A space after comma.

Also, I'd prefer OOP method:

``` php
(new DateTime('next monday'))->format('d-m-Y')
```

ameenross commented on src/LocationClient.php:
```diff
@@ -0,0 +1,31 @@
+    {
+        return $this->__soapCall('GetNearestLocations', [$getNearestLocations]);
+    }
+}
```
Newline at end of file

ameenross commented on src/LocationClient.php:
```diff
@@ -0,0 +1,31 @@
+     *     The complex types used by this client.
+     */
+    protected $classes = [
+        'GetNearestLocation',
```
I think there should be a lot more, like `Message`.

ameenross commented on src/Postnl.php:
```diff
@@ -315,6 +315,36 @@ public function currentStatus($barcode)
+     * @throws ComplexTypes\CifException
+     * @throws SoapFault
+     */
+    public function getNearestLocation($postalCode, $allowSundaySorting = "false", $deliveryDate = null) {
```
Opening brace on next line

danslo commented:
Any update on this PR? I'll probably need this functionality relatively soon. I can update the styling issues if you haven't started on it yet.

ameenross commented:
You're welcome to. You'd have to create a branch off of @Jomaanro's branch though.
